spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,96 @@
       end
     end
   end
+
+  describe ".get_col_info" do
+    it "return column info for virtual field" do
+      field = "VmInfra-archived"
+      col_info = described_class.get_col_info(field)
+      expect(col_info[:data_type]).to eq(:boolean)
+      expect(col_info[:excluded_by_preprocess_options]).to eq(false)
+      expect(col_info[:format_sub_type]).to eq (:boolean)
+      expect(col_info[:include]).to eq({})
+      expect(col_info[:tag]).to eq(false)
+      expect(col_info[:virtual_column]).to eq(true)
+      expect(col_info[:virtual_reflection]).to eq(false)
+    end
+
+    it "return column info for managed-field" do
+      field = "managed-location"
+      col_info = described_class.get_col_info(field)
+      expect(col_info[:data_type]).to eq(:string)
+      expect(col_info[:excluded_by_preprocess_options]).to eq(false)
+      expect(col_info[:format_sub_type]).to be_nil
+      expect(col_info[:include]).to eq({})
+      expect(col_info[:tag]).to eq(true)
+      expect(col_info[:virtual_column]).to eq(false)
+      expect(col_info[:virtual_reflection]).to eq(false)
+    end
+
+    it "return column info for model.managed-field" do
+      field = "VmInfra.managed-operations"
```

Comment:
It might be better to call this a tag as a field is something different in the context of MiqExpression. I haven't delved into these too much yet but I believe the structure is `Model.namespace-category` - @gtanzillo is this correct?